core/src/main/java/com/sequenceiq/cloudbreak/service/stack/StackService.java

```diff
@@ -434,6 +437,30 @@ public void removeInstance(IdentityUser user, Long stackId, String instanceId) {
         flowManager.triggerStackRemoveInstance(stackId, instanceMetaData.getInstanceGroupName(), instanceMetaData.getPrivateId());
     }
 
+    public void removeInstances(IdentityUser user, Long stackId, Set<String> instanceIds) {
+        Stack stack = get(stackId);
```

Comment:
The get method checks for permissions (authorizationService.hasReadPermission(stack);), but this method should be extended with hasWritePermission instead of manually checking the owner and public in account field below.